### PR TITLE
terminfo: add header guard, stdint.h for int8_t

### PR DESCRIPTION
**Pull request by janlazo**
Lint fixes for single-include test.

Lint errors from #7981 are annoying me so I'm addressing them here.

### CODE REVIEW

**Comment by justinmk** [comment]
good catch, thanks!

> Lint errors from #7981 are annoying me so I'm addressing them here.

How about adding `// NOLINT` to the generated long lines?

**Comment by janlazo** [comment]
@justinmk I'm updating `terminfo_defs.h` manually so I don't want to add the `NOLINT` comments for each terminfo definition.

@mhinz Can you do it as follow-up for #7981?

File: scripts/update_terminfo.sh [hunk 1]
```diff
@@ -61,8 +61,13 @@ cat > "$target" <<EOF
 // it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
```
Comment (justinmk) [inline scripts/update_terminfo.sh]: I also noticed that none of our headers have this PVS message. I guess it's not needed. Right @ZyX-I ?

Comment (ZyX-I) [reply scripts/update_terminfo.sh]: PVS (currently) operates on C files only, it does not analyze header files (side from the fact that PVS runs preprocessor first which dumps used headers inside C files) and thus does not need headers there.

**Pull request merged**